source/dk_device.cpp

```diff
@@ -257,3 +257,21 @@ void ObjBase::freeMem(void* mem) const noexcept
 {
 	return m_device->freeMem(mem);
 }
+
+uint64_t dkDeviceGetCurrentTimestamp([[maybe_unused]] DkDevice obj) {
+	u64 ts;
+	Result rc = nvGpuGetTimestamp(&ts);
+	return R_SUCCEEDED(rc) ? ts : 0;
+}
+
+uint64_t dkDeviceTimestampToNs([[maybe_unused]] DkDevice obj, uint64_t ts) {
+	return (ts * 625) / 384;
+}
+
+uint64_t dkDeviceNsToTimestamp([[maybe_unused]] DkDevice obj, uint64_t tick) {
+	return (tick * 384) / 625;
+}
```

Comment:
These helper functions should be in the header, marked as DK_CONSTEXPR, and without a DkDevice parameter.

Reply:
> without a DkDevice parameter

Same for `dkDeviceGetCurrentTimestamp(InNs)`? I wasn't so sure about which signature was best. It doesn't need the DkDevice param, but nvn has it (probably because it has to support several platforms) so that's what I went with.

Reply:
`dkDeviceGetCurrentTimestamp(InNs)` can be left as a "member function" of DkDevice as it is a non-trivial function that depends on OS functionality. While it is true that the current implementation does not require any data stored within DkDevice itself, it might not always be the case (i.e. we might want to refactor libnx's nv wrappers at any point).

Reply:
Got it